processor/src/host/debug.rs

```diff
@@ -1,175 +1,170 @@
         },
     }
 }
 
 // HELPER FUNCTIONS
 // ================================================================================================
 
-struct Printer {
-    clk: RowIndex,
-    ctx: ContextId,
-    fmp: u32,
-}
-
-impl Printer {
-    fn new(clk: RowIndex, ctx: ContextId, fmp: u64) -> Self {
-        Self { clk, ctx, fmp: fmp as u32 }
-    }
+/// Prints the number of stack items specified by `n` if it is provided, otherwise prints
+/// the whole stack.
+fn print_vm_stack(process: &ProcessState, n: Option<usize>) {
+    let stack = process.get_stack_state();
 
-    /// Prints the number of stack items specified by `n` if it is provided, otherwise prints
-    /// the whole stack.
-    fn print_vm_stack(&self, process: &ProcessState, n: Option<usize>) {
-        let stack = process.get_stack_state();
+    // determine how many items to print out
+    let num_items = core::cmp::min(stack.len(), n.unwrap_or(stack.len()));
 
-        // determine how many items to print out
-        let num_items = core::cmp::min(stack.len(), n.unwrap_or(stack.len()));
-
-        // print all items except for the last one
-        println!("Stack state before step {}:", self.clk);
-        for (i, element) in stack.iter().take(num_items - 1).enumerate() {
-            println!("├── {i:>2}: {element}");
-        }
-
-        // print the last item, and in case the stack has more items, print the total number of
-        // un-printed items
-        let i = num_items - 1;
-        if num_items == stack.len() {
-            println!("└── {i:>2}: {}\n", stack[i]);
-        } else {
-            println!("├── {i:>2}: {}", stack[i]);
-            println!("└── ({} more items)\n", stack.len() - num_items);
-        }
+    // print all items except for the last one
+    println!("Stack state before step {}:", process.clk());
+    for (i, element) in stack.iter().take(num_items - 1).enumerate() {
+        println!("├── {i:>2}: {element}");
     }
 
-    /// Prints length items from the top of the  advice stack. If length is 0 it returns the whole
-    /// stack.
-    fn print_vm_adv_stack(&self, advice_provider: &AdviceProvider, length: usize) {
-        let stack = advice_provider.peek_stack(length);
+    // print the last item, and in case the stack has more items, print the total number of
+    // un-printed items
+    let i = num_items - 1;
+    if num_items == stack.len() {
+        println!("└── {i:>2}: {}\n", stack[i]);
+    } else {
+        println!("├── {i:>2}: {}", stack[i]);
+        println!("└── ({} more items)\n", stack.len() - num_items);
+    }
+}
 
-        // we may have less elements than requested
-        let num_items = stack.len();
-        if num_items == 0 {
-            println!("Advice Stack empty before step {}.", self.clk);
-            return;
-        };
+/// Prints length items from the top of the  advice stack. If length is 0 it returns the whole
+/// stack.
+fn print_vm_adv_stack(process: &ProcessState, length: usize) {
```

Comment:
nit: since we are refactoring anyways, could we rename `length` to `n` and pass it as `Option<usize>` to be consistent `print_vm_stack()`?

Reply:
It's not much different to use `Option` and IMO is slightly less clear. 
```rust
fn print_vm_adv_stack(process: &ProcessState, n: Option<usize>) {
    let mut stack = process.advice_provider().stack();
    let num_items = if let Some(stack_len) = n {
        min(stack.len(), stack_len)
    } else {
        stack.len()
    };
    stack = &stack[..num_items];
```

we also have to add 
```rust
let n = (length != 0).then(|| length as usize);
```
to the caller. 

I'll change the argument to `n` but leave the rest as is

Reply:
I think you'd typically do `Option<NonZeroUsize>` to make it clear that _if_ there is a length being specified, that it is non-zero.